testing/unit_tests/test_automate_training.py

```diff
@@ -0,0 +1,697 @@
+"""Unit Test for Automate Training
```

Comment:
This is only a suggestion, but I noticed the mocking of the config makes this file quite lengthy. Would it maybe make more sense to place the mocked data in another file and import it into this one to separate the test logic from the mock data?

Reply:
Yah I was actually wondering that myself, that's a good point! I should probably put it all in a json file or something and save it in the `data-testing` repo.

Reply:
https://github.com/ivadomed/data-testing/pull/10

Reply:
So it looks like it isn't too trivial to parametrize tests from data that is loaded, especially because we download that data in the `setup_function`. Basically, if I load that data outside the `setup_function`, the data won't have been downloaded yet. And if we load it in the `setup_function`, it isn't global and the tests don't have access to it.